### ISSUE TITLE
Missing translation possibility for social network

### ISSUE THREAD

**Issue by lonesomewalker** [chamilo/chamilo-lms#2204]
### Current behavior / Resultado actual / Résultat actuel
When posting some stuff at social network, date/time is only shown in english.
No translation available.
(example: 7 minutes ago)

### Expected behavior / Resultado esperado / Résultat attendu
Also this part of Chamilo should have strings which can be translated.

### Chamilo Version / Versión de Chamilo / Version de Chamilo
1.11.4

**Comment by lonesomewalker** [comment]
Digging further...

Seems the jquery-timeago doesn't load the locales.
Referenced in main/inc/lib/template.lib.php and main/social/profile.php.
But both not working.

It would be possible to add in main/inc/lib/template.lib.php:
"$bowerJsFiles[] = 'jquery-timeago/locales/jquery.timeago.'.$isoCode.'.js';" after line 761

But it seems to be an issue with timeago itself, since:
[https://github.com/rmm5t/jquery-timeago/issues/323](https://github.com/rmm5t/jquery-timeago/issues/323)

Also seen in generated page source code that the jQuery timeago is loaded 2 times...?

**Comment by ywarnier** [comment]
Assets are dynamically generated by our JS dependencies manager (bower) while api_get_js() loads files from the more persistent main/inc/lib/ directory.
Feel free to submit pull requests to improve the documentation. This is in your reach.

**Comment by ywarnier** [comment]
Just as a precision, in the code you mentioned, what we do is we load the generic timeago lib, and then we check if the localized version (basically the translations JSON) exists for the current language. If it does, we load it. Otherwise we don't. This means that if you want the German translations to be available, you need the file web/assets/jquery-timeago/locales/jquery.timeago.de.js to exist. I see in the latest code that it exists, but maybe that was an addition after 1.11.4. I haven't checked.

**Comment by lonesomewalker** [comment]
No, it is available, but it doesn't get loaded... So i think the if statement doesn't get "true". Will debug this later to see what's happening.

**Comment by lonesomewalker** [comment]
Okay, as i thought:
`$timeAgoLocaleDir = $javascriptDir.'jquery-timeago/locales/jquery.timeago.'.$locale.'.js';`

becomes false, because it is the wrong directory...
(it resolves to: /main/inc/lib/javascript/jquery-timeago/locales/jquery.timeago.de.js).

So, **no enhancement**, it is a **bug** which can be reproduced.

**Comment by ywarnier** [comment]
Fair enough.
Did you try changing it in the code on your platform already to see if that works?

**Comment by lonesomewalker** [comment]
Sure :-) works as expected, now all the timestamps get translated.

**Comment by ywarnier** [comment]
Fancy sending your own pull request? Or too much to learn for such a little change?

**Comment by lonesomewalker** [comment]
Didn't i do it right?
Or do i have to pull to chamilo:1.11.x ?

**Comment by ywarnier** [comment]
A pull request is its own big thing, the first time you do it (you have to start from your own fork of the project on Github). I suggest you check our Contributing.md file for explanations: https://github.com/chamilo/chamilo-lms/blob/1.11.x/CONTRIBUTING.md

**Comment by ywarnier** [comment]
Actually, sorry, this is more of this link: https://guides.github.com/activities/contributing-to-open-source/ (which is included in CONTRIBUTING.md but more focused on your case).

**Comment by ywarnier** [comment]
You're right, I missed it. In general, if you report an issue, then you can associate the commit in your PR with the issue by simply mentioning #2204. That would have attached your commit to this report, but that's OK for now. I'll check it and merge it (if alright) tomorrow.

**Comment by jmontoyaa** [comment]
The change in #2207  should be send to branch 1.11.x not yet to master.

**Comment by lonesomewalker** [comment]
But it is related to both.
It is a wrong path so do i have to commit it to both?

**Comment by jmontoyaa** [comment]
@lonesomewalker  then you have to send the commit to both branches
